Shuffle.cs

```diff
@@ -514,22 +516,98 @@ private void MakeSpoilerLog()
             {
                 ItemList.RemoveRange(WF_Map, ST_Key4 - WF_Map + 1);
             }; */
+
+
             ItemList.RemoveAll(u => u.ReplacesItemId == -1);
-            LogFile.WriteLine("--------------Item------------------------------Destination-----------");
+            LogFile.WriteLine("-------------------Item------------------------------------Location-----------------");
             for (int i = 0; i < ItemList.Count; i++)
             {
-                LogFile.WriteLine(Items.ITEM_NAMES[ItemList[i].ID].PadRight(32, '-') + "---->>" + Items.ITEM_NAMES[ItemList[i].ReplacesItemId].PadLeft(32, '-'));
+                LogFile.WriteLine(Items.ITEM_NAMES[ItemList[i].ID].PadRight(40, '-') + "->>-" + Items.ITEM_NAMES[ItemList[i].ReplacesItemId].PadLeft(40, '-'));
             };
-            LogFile.WriteLine("");
-            LogFile.WriteLine("-----------Destination------------------------------Item--------------");
-            ItemList.Sort((i, j) => i.ReplacesItemId.CompareTo(j.ReplacesItemId));
+
+            LogFile.WriteLine();
+            LogFile.WriteLine();
+
+            LogFile.WriteLine("-------------------Item------------------------------------Location-----------------");
             for (int i = 0; i < ItemList.Count; i++)
             {
-                LogFile.WriteLine(Items.ITEM_NAMES[ItemList[i].ReplacesItemId].PadRight(32, '-') + "<<----" + Items.ITEM_NAMES[ItemList[i].ID].PadLeft(32, '-'));
+                LogFile.WriteLine(Items.ITEM_NAMES[ItemList[i].ReplacesItemId].PadRight(40, '-') + "->>-" + Items.ITEM_NAMES[ItemList[i].ID].PadLeft(40, '-'));
             };
             LogFile.Close();
         }
 
+        private void MakeHTMLSpoilerLog()
+        {
+            var settingsString = EncodeSettings();
+
+            var directory = Path.GetDirectoryName(Settings.OutputROMFilename);
+            var filename = $"{Path.GetFileNameWithoutExtension(Settings.OutputROMFilename)}_SpoilerLog.html";
+
+            StreamWriter LogFile = new StreamWriter(Path.Combine(directory, filename));
+
+            LogFile.WriteLine("<html>");

```

Comment:
I feel like the HTML should be in a Resource file. Probably don't need to go as far as using a compiler, but this should definitely be cleaned up.

Reply:
I'm not sure exactly what you mean by this.

Reply:
i.e. have a resource that's like SpoilerLogHtmlTemplate
```<html><head><style>
...
<table ...>
<thead></thead>
<tbody>{0}</tbody>
<table>
...
</html>
```
and another that's like SpoilerLogTableRowTemplate
```
<tr><td>{0}</td><td class="spoiler">{1}</td></tr>
```
etc.